calculator/src/main/java/com/market/banica/calculator/service/ProductServiceImpl.java

```diff
@@ -55,6 +56,9 @@ public void createProduct(String newProductName, String unitOfMeasure,
         LOGGER.debug("In createProduct method with parameters: newProductName {}, unitOfMeasure {} and ingredientsMap {}"
                 , newProductName, unitOfMeasure, ingredientsMap);
 
+        DataValidator.validateIncomingData(newProductName);
```

Comment:
There is repeated validation of the same fields in JMXServiceImpl.createProduct. Decide where to validate the fields in JMXServiceImpl or ProductServiceImpl. I think the best is to validate only ProductServiceImpl, because as far as I remember all other services (in this case JMXServiceImpl and the rest service) come down to ProductServiceImpl and this will avoid adding validations in each of them.

Reply:
It's already fixed. I remove the unnecessary data validation in JMXServiceImpl.